enhancements/insights/insights-config-api.md

```diff
@@ -0,0 +1,335 @@
+	// gathering of the Insights archive and its uploading to the ingress.
+	GatheringConfig *GatheringConfig `json:"gatheringConfig,omitempty"`
+}
+
+type InsightsStatus struct {
+}
+
+type GatheringConfig struct {
+	// DataPolicy allows user to enable additional global obfuscation of the IP addresses and base domain
+	// in the Insights archive data.
+	// +kubebuilder:default=NoPolicy
+	DataPolicy DataPolicy `json:"dataPolicy"`
+	// ForceGatherReason enables user to force Insights data gathering by setting a new reason.
+	// When there is some gathering in the progress then it is interrupted.
+	// When all the gatherers are deactivated by the `DisabledGatherers`, nothing happens.
+	// When the forced gathering is finished then the value is cleared.
+	ForceGatherReason string `json:"forceGatherReason"`
+	// List of gatherers to be excluded from the gathering. All the gatherers can be disabled by providing "all" value.
+	// If all the gatherers are disabled, the Insights operator does not gather any data.
+	// The particular gatherers IDs can be found at https://github.com/openshift/insights-operator/blob/master/docs/gathered-data.md.
+	// An example of disabling gatherers looks like this: `disabledGatherers: ["clusterconfig/machine_configs", "workloads/workload_info"]`
+	DisabledGatherers []string `json:"disabledGatherers"`
+}
+
+const (
+	// No data obfuscation
+	NoPolicy DataPolicy = "NoPolicy"
+	// IP addresses and cluster domain name is obfuscated
+	IPsAndClusterDomainPolicy DataPolicy = "IPsAndClusterDomainPolicy"
+)
+
+type DataPolicy string
+```
+
+The proposal for the new `insights.operator.openshift.io` resource is:
+
+```go
+type Insights struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+
+	// spec is the specification of the desired behavior of the Insights
+	// +kubebuilder:validation:Required
+	// +required
+	Spec InsightsSpec `json:"spec"`
+
+	// status is the most recently observed status of the Insights operator
+	// +optional
+	Status InsightsStatus `json:"status"`
+}
+
+type InsightsSpec struct {
+	OperatorSpec `json:",inline"`
+}
+
+type InsightsStatus struct {
+	// GatheringStatus provides basic information about the last Insights gathering.
+	GatheringStatus *GatheringStatus `json:"gatheringStatus,omitempty"`
+	// ReportStatus provides general Insights analysis results.
+	ReportStatus *ReportStatus `json:"reportStatus,omitempty"`
+}
+
+type GatheringStatus struct {
+	// LastGatherTime is the last time when Insights gathering finished.
+	LastGatherTime metav1.Time `json:"lastGatherTime,omitempty"`
+	// LastGatherReason provides last known reason of gathering. This is helpful
+	// especially when gathering was forced by user
+	LastGatherReason string `json:"lastGatherReason,omitempty"`
+	// StartGatherTime is the time when gathering started. The value is nil
+	// when there is no gathering in progress.
+	StartGatherTime metav1.Time `json:"startGatherTime,omitempty"`
```

Comment:
If you want to make it nil, you need to make it a pointer and make it `// +nullable`, iow:
```golang
// +nullable
StartGatherTime *metav1.Time `json:"startGatherTime,omitempty"`
```

Reply:
Alternatively you can change the description to say it will be 0, metav1 has `IsZero` function check, and that's fine as well. Just make it clear, because the current wording is confusing.

Reply:
Updated with 0 as default value